docs/contributor/draft/rate-limit-api.yaml

```diff
@@ -0,0 +1,54 @@
+# Since combining global and global rate limit would make API more complex
+# I would avoid doing so, and in case of extending Istio Module capabilities
+# to global rate limit, I'd introduce new CRD. This way we would make use of our API easier.
+# User would create LocalRateLimit and GlobalRateLimit custom objects separately
+# One LocalRateLimit per workload for which rate limit is applied.
+# This way it's easier for user to keep service related resources in one place.
+# Also avoiding long yamls in case of big clusters with big amounts of workloads
+
+# This is how we could introduce it for now covering simple rate limiting without any specific selectors
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: LocalRateLimit
+metadata:
+  name: httpbin-local-rate-limit
+  namespace: default
+spec:
+  workloadSelector:
```

Comment:
In case of workloadSelector being skipped it applies to all workloads within the namespace, and in addition when it's in the istio-system (root config ns) it applies to all workloads across all namespaces. Similar to the EnvoyFilter way of working.